### ISSUE TITLE
Snap Notification Rate Limits

### ISSUE THREAD

**Issue by mario-christopher** [MetaMask/metamask-docs#1158]
We've had a few builders ask us why they see Rate Limit messages show up in the Notification Snaps.
Unfortunately our docs do not describe the feature fully, nor mention that there are Rate Limits for notifications.  It would be helpful to document this feature.

Reading from the code, I got this information (thanks @FrederikBolding )
https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/metamask-controller.js#L1222-L1257

TLDR:
```
Native notifications: 2 calls per 5 minutes per snap
In app notifications: 5 calls per minute per snap
```
